docker/Dockerfile

```diff
@@ -1,17 +1,20 @@
-FROM mcr.microsoft.com/dotnet/aspnet:5.0-alpine
+FROM mcr.microsoft.com/dotnet/sdk:5.0-alpine as builder
 
```

Comment:
General comment: could you please add a build argument to still allow the current use case of using a pre-built directory to build the container?  (could be "selfContained" for the new use case being introduced).  In some cases, we need to build, then run tests (not container based today), then copy the exact binaries that were tested on the container.

Reply:
It isn't possible to do a conditional COPY (at least not without hacky tricks).

So what you need is to build and test binaries outside of a container, and then COPY those into an image, as currently on main. What will that image then be used for?

This proposal is to have a Dockerfile which can be built by anyone, anywhere, without any dependency on a specific environment. This would then also be the published image, so this is reproducible.

You can of course create a separate Dockerfile for the process you describe.

Reply:
Ok, if this is messy and non-standard, then it makes sense to have the standard Dockerfile be the default, which should be the version with your changes.  I'll try the approach of copying the binaries out of the container to sanity test them in CI before publishing (the image built in our pipeline is the one pushed to the official registry on DockerHub ([https://hub.docker.com/_/microsoft-restlerfuzzer-restler](https://hub.docker.com/_/microsoft-restlerfuzzer-restler)).

Reply:
Excellent, sounds good.